docs/config-file/v2.rst

```diff
@@ -297,43 +294,104 @@ The path to the Conda environment file, relative to the root of the project.
 :Type: ``path``
 :Required: ``true``
 
-build
-~~~~~
+build (beta specification)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. warning::
+
+   This functionality is in beta. If you find any inconsistencies
+   or have feedback, please `open an
+   issue <https://github.com/readthedocs/readthedocs.org/issues/>`_.
 
 Configuration for the documentation build process.
+This allows you to specify the base Read the Docs image
+used to build the documentation,
+and control the versions of several tools:
+Python, Node.js, Rust, and Go.
 
 .. code-block:: yaml
 
    version: 2
 
    build:
-     image: latest
-     apt_packages:
-       - libclang
-       - cmake
+     os: ubuntu-20.04
+     tools:
+       python: "3.9"
+       nodejs: "16"
+       rust: "1.55"
+       golang: "1.17"
 
-   python:
-     version: "3.7"
+build.os
+````````
+
+The Docker image used for building the docs.
+Image names refer to the operating system Read the Docs uses to build them.
 
 .. note::
 
-   Check out the :ref:`new-build` below for an alternative method
-   that is more flexible.
+   Arbitrary Docker images are not supported.
+
+:Type: ``string``
+:Options: ``ubuntu-20.04``
+:Required: ``true``
 
-build.image
+build.tools
 ```````````
 
-The Docker image used for building the docs.
+Version specifiers for each tool. It must contain at least one tool.
+
+:Type: ``dict``
+:Options: ``python``, ``nodejs``, ``rust``, ``golang``
+:Required: ``true``
+
+build.tools.python
+``````````````````
+
+Python version to use.
+You can use several interpretes and versions, from CPython, PyPy, Miniconda, and Mamba.
+
+.. note::
+
+   If you use Miniconda3 or Mambaforge, you can select the Python version
+   using the ``environment.yml`` file. See our :doc:`/guides/conda` guide
+   for more information.
 
 :Type: ``string``
-:Options: ``stable``, ``latest``
-:Default: ``latest``
+:Options:
+  - ``2.7``
+  - ``3``
```

Comment:
Should we mention what `3` means here?

Reply:
Good idea. "Last stable version of CPython"?

Reply:
Done ✔️